binlogreplication/binlog_replica_applier.go

```diff
@@ -1291,17 +1296,25 @@ varsLoop:
 
 // getTableSchema returns a sql.Schema for the case-insensitive |tableName| in the database named
 // |databaseName|, along with the exact, case-sensitive table name.
-func getTableSchema(ctx *sql.Context, engine *gms.Engine, tableName, databaseName string) (sql.PrimaryKeySchema, string, error) {
-	database, err := engine.Analyzer.Catalog.Database(ctx, databaseName)
-	if err != nil {
-		return sql.PrimaryKeySchema{}, "", err
-	}
-	table, ok, err := database.GetTableInsensitive(ctx, tableName)
-	if err != nil {
-		return sql.PrimaryKeySchema{}, "", err
-	}
-	if !ok {
-		return sql.PrimaryKeySchema{}, "", fmt.Errorf("unable to find table %q", tableName)
+func (a *binlogReplicaApplier) getTableSchema(ctx *sql.Context, engine *gms.Engine, tableName, databaseName string) (sql.PrimaryKeySchema, string, error) {
+	key := strings.ToLower(databaseName + "." + tableName)
```

Comment:
`strings.ToLower(catalog.ConnectIdentifiersANSI(databaseName, tableName))` to avoid collisions between `"a.b"."c"` and `"a"."b.c"`?

Reply:
Or using a `struct` as the key - similar to `DeltaController.tables`.

Reply:
Thank you for comment. I went with second approach. Fixed and force pushed.